packages/squiggle-lang/src/rescript/Distributions/GenericDist/GenericDist.res

```diff
@@ -186,6 +186,49 @@ module AlgebraicCombination = {
     ->E.R2.fmap(r => DistributionTypes.SampleSet(r))
   }
 
+  /*
+     It would be good to also do a check to make sure that probability mass for the second
+     operand, at value 1.0, is 0 (or approximately 0). However, we'd ideally want to check 
+     that both the probability mass and the probability density are greater than zero.
+     Right now we don't yet have a way of getting probability mass, so I'll leave this for later.
+ */
+  let getLogarithmInputError = (t1: t, t2: t, ~toPointSetFn: toPointSetFn): option<error> => {
+    let firstOperandIsGreaterThanZero =
+      toFloatOperation(t1, ~toPointSetFn, ~distToFloatOperation=#Cdf(1e-10)) |> E.R.fmap(r =>
+        r > 0.
+      )
+    let secondOperandIsGreaterThanZero =
+      toFloatOperation(t2, ~toPointSetFn, ~distToFloatOperation=#Cdf(1e-10)) |> E.R.fmap(r =>
+        r > 0.
+      )
+    let items = E.A.R.firstErrorOrOpen([
+      firstOperandIsGreaterThanZero,
+      secondOperandIsGreaterThanZero,
+    ])
+    switch items {
+    | Error(r) => Some(r)
+    | Ok([true, _]) =>
+      Some(LogarithmOfDistributionError("First input must completely greater than 0"))
+    | Ok([false, true]) =>
+      Some(LogarithmOfDistributionError("Second input must completely greater than 0"))
+    | Ok([false, false]) => None
+    | Ok(_) => Some(Unreachable)
```

Comment:
If you use `E.R.merge` rather than `E.A.R.firstErrorOrOpen` you won't need this line

Reply:
I don't think this works. If I do E.R.Merge, I won't know which one had the error in question. 

It's kind of confusing because there are two very different types of errors here.